test/cpp/tensorexpr/test_llvm.cpp

```diff
@@ -801,7 +801,7 @@ void testLLVMSimpleMath01() {
       "f", {{N, "i"}}, [](const VarHandle& i) { return cast<float>(i * i + 1); });
   Schedule sch = Schedule::make({tensor});
   Stmt* stmt = sch.Lower();
-  Buffer f_buf(VarHandle(tensor->function()->func_var()), kFloat32, {N});
+  Buffer f_buf(VarHandle(tensor->function()->func_var(tensor->output_index())), kFloat32, {N});
```

Comment:
We need a test to show a function with multiple output tensors actually work.

Reply:
I didn't implement codegen/schedules for multi-output functions in this PR. However, this PR in isolation is a net improvement over the current state: instead of pretending that we're support ing multi-output functions and silently behaving incorrectly in a case if someone tries to use it, we would fail loudly with assertion.

Do you mind if this PR gets committed without codegen/schedule part and I'll follow up on that in a later PR?